src/WebJobs.Script/Binding/DocumentDBBinding.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding
+{
+    internal class DocumentDBBinding : FunctionBinding
+    {
+        private BindingDirection _bindingDirection;
+
+        public DocumentDBBinding(ScriptHostConfiguration config, string name, string databaseName, string collectionName, bool createIfNotExists, FileAccess access, BindingDirection direction) :
+            base(config, name, "documentdb", access, false)
+        {
+            this.DatabaseName = databaseName;
+            this.CollectionName = collectionName;
+            this.CreateIfNotExists = createIfNotExists;
+            _bindingDirection = direction;
+        }
+
+        public string DatabaseName { get; private set; }
+
+        public string CollectionName { get; private set; }
+
+        public bool CreateIfNotExists { get; private set; }
+
+        public override bool HasBindingParameters
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override CustomAttributeBuilder GetCustomAttribute()
+        {
+            PropertyInfo[] props = new[]
+           {
+                typeof(DocumentDBAttribute).GetProperty("DatabaseName"),
+                typeof(DocumentDBAttribute).GetProperty("CollectionName"),
+                typeof(DocumentDBAttribute).GetProperty("CreateIfNotExists")
+            };
+
+            object[] propValues = new object[]
+            {
+                this.DatabaseName,
+                this.CollectionName,
+                this.CreateIfNotExists
+            };
+
+            ConstructorInfo constructor = typeof(DocumentDBAttribute).GetConstructor(System.Type.EmptyTypes);
+
+            return new CustomAttributeBuilder(constructor, new object[] { }, props, propValues);
```

Comment:
nit: Since we're using typeof(DocumentDBAttribute) a few times in this method, it would be a good idea to get the type at the beginning and reuse it to avoid the additional calls.